src/app/services/covid-data-api.service.ts

```diff
@@ -18,72 +15,28 @@ export const COVID_DATA_API_SUB_SERVICE = new InjectionToken<
   providedIn: 'root',
 })
 export class CovidDataApiService {
+  summaryDataObserver: Subject<CovidDataPoint[]> = new Subject<CovidDataPoint[]>();
   constructor(
     // Inspired by (https://stackoverflow.com/a/35916788)
     // Each api service must have its own provider registered in app.module for the InjectionToken
     @Inject(COVID_DATA_API_SUB_SERVICE)
     private apiServices: CovidDataApiSubService[]
-  ) {}
+  ) {
+    this.getCovidSummaryData();
+  }
 
   /**
    * Calls all api services and returns the first non-error response
-   * If all lead to errors, returns an empty observable
-   *
-   * @param {Observable<CovidDataPoint>} method The method to be called
+   * If all lead to errors, prints an error message to the console
    */
-  private getResultFromOneApiFor(
-    country?: Country
-  ): Observable<CovidDataPoint> {
-    let errorCount = 0; // keep track of number or errored out requests for later
-
-    return race( // only need result from one of the apis
-      this.apiServices.map<Observable<CovidDataPoint>>((api, i, apis) =>
-        new GetLatestDataCommand(api, country).call().pipe(
-          catchError((err) => {
-            errorCount += 1;
-            this.handleError(err);
-            return (errorCount === apis.length)
-            ? throwError(noGoodApiResponseError())
-            // If all other subservices errored out, return an error observable to end the race
-            : NEVER; // Use NEVER so that an erroring out request doesn't end the race
-          })
-        )
-      )
-    )
-  }
-
-  getLatestGlobalData(): Observable<CovidDataPoint> {
-    return this.getResultFromOneApiFor();
-  }
-
-  getLatestCountryData(country: Country): Observable<CovidDataPoint> {
-    return this.getResultFromOneApiFor(country);
-  }
-
-  handleError(err: Error): void {
-    // TODO: replace with something better when we have a error handler/logger implemented
-    console.log(err);
+  private getCovidSummaryData(): void {
+    const callFunction = this.apiServices.reduceRight<(e?: any) => void>(
+      (fallback: () => void, api: CovidDataApiSubService) => () => api.call(
+        result => this.summaryDataObserver.next(result),
+        fallback
+      ),
+      () => console.error('All api sub-services responded with errors')
+    );
```

Comment:
Couldn't get what you mean't in #19. But this does look a lot cleaner. 

Still not sure I fully understand it, since I'm used to using `reduce`/`reduceRight` for [accumulating over arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/ReduceRight).

In this case, the `console.error(...)` function is assigned to `fallback` as the initial value (and is never replaced).
Then you loop through each `api` and return the result of each of them to subscribers in the `next(...)`. Could the fact that its doing it to all apis be a bug? Wouldn't it replace the results from the quickest apis with the slowest ones?

If not, then this is doing the same as
```
this.apiServices.forEach(api => api.call(success, fail));

success = () => this.summaryDataObserver.next(result);

fail = () => console.error('All api sub-services responded with errors');
```
is it not?

Reply:
No, forEach would call ALL apis, the code above calls the first api in the array, if it fails calls the second etc. and if all fail prints the error message.
Your code calls all apis and prints the error multiple times.

To clarify, if we have an array `[api1, api2]`, the `callFunction` will become
```
callFunction = () => api1.call(
   result => this.summaryDataObserver.next(result), // api1 success
   () => api2.call( // api1 error
     result => this.summaryDataObserver.next(result), // api1 error api2 success
     () => console.error('All api sub-services responded with errors'); // api1 and api2 error
   )
```
so as you can see the call to api2 is inside the error callback of api1. Hope it's clearer now.

Reply:
Ah ok, so this is building a chain of error fallbacks from the last api up. Where the last one is that `console.error` line. 
And each time you go through an api in the reduceRight loop, the `fallback` (which is the reduce accumulator) is set as the call to the api it just looped.

Part of my confusion in the first comment was due to me missing the `() =>` before `api.call` in the reduceRight as well as the reduction on functions which I don't think I've used before. 

But it now does make sense. Cool stuff!